src/DynamoCore/Models/WorkspaceModel.cs

```diff
@@ -668,19 +668,25 @@ public AnnotationModel AddAnnotation(string text, Guid id)
         {
             var selectedNodes = this.Nodes == null ? null:this.Nodes.Where(s => s.IsSelected);
             var selectedNotes = this.Notes == null ? null: this.Notes.Where(s => s.IsSelected);
-
-            var annotationModel = new AnnotationModel(selectedNodes, selectedNotes)
+            //No Groups should be selected when creating a new group. This is to avoid creating multiple groups
+            //on the selected nodes.
+            var selectedAnnotation = this.Annotations == null ? null : this.Annotations.Where(s => s.IsSelected);
+            if (selectedAnnotation != null && !selectedAnnotation.Any())
```

Comment:
Sure @pboyer - I am checking on that behavior as well. There is a different task to check for empty groups and that might handle this I guess. I will check on that.